### PR TITLE
Added a check in GetMeteredVariables so that invalid Meter:Custom do not...

### PR DESCRIPTION
**Pull request by JasonGlazer**
Added a check so that invalid Meter:Custom do not cause a crash and instead show a warning message in the ERR file. This is related to #4631

### CODE REVIEW

**Comment by JasonGlazer** [comment]
I checked a bunch of the failed tests and they all seem to be EIO changes that are just changes to wording rather than changes to values. I am not sure why these were shown as changes to this pull request since I don't think I changed any wording of messages in the EIO files.

**Comment by JasonGlazer** [comment]
@mjwitte and @Myoldmopar Is it possible that the CI tests are picking up some other changes and showing them as diffs to these files?

**Comment by mjwitte** [comment]
@JasonGlazer It's because develop is ahead of your branch.  If you want to clear those, you need to pull the latest develop and merge it into your branch, then push it back up.

**Comment by mjwitte** [comment]
@JasonGlazer Yesterday or so, I merged in a bunch of filed name changes that Rich did for unitary system and sizing.  So, that's causing differences in lost of eio output lines.

**Comment by JasonGlazer** [comment]
Having the CI system compare against a different version than the source than was the basis for the changes to each fix means that each developer needs to figure out what is going on this time. Maybe the CI should change what the base case is.

**Comment by mjwitte** [comment]
And if we are open to that, I would suggest that it include, or only include, annual simulations with diffs on daily or monthly results.

**Comment by JasonGlazer** [comment]
The current system does not provide the individual developer an understanding if failed tests mean the code is broken and because of this, all the developer can assert is that it *should* work or not. That seems to defeat one of the goals of CI. 

Can we at least add an additional CI test suite on one platform that compares the new code to the code used as a basis for the new code? This would provide the developer an understanding that the code works or not.

File: tst/EnergyPlus/unit/OutputProcessor.unit.cc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	GetMeteredVariables( TypeOfComp, NameOfComp, VarIndexes, VarTypes, IndexTypes, UnitsStrings, ResourceTypes, EndUses, Groups, Names, NumFound );
+
+	EXPECT_EQ( 0, NumFound );
+
+	NumOfRVariable = 2;
+	NameOfComp = "OUTSIDELIGHTS";
+	RVar.allocate();
+
+	RVar().MeterArrayPtr = 1;
+	RVariableTypes( 1 ).KeyNameOnlyUC = NameOfComp;
```
Comment (jasondegraw) [inline tst/EnergyPlus/unit/OutputProcessor.unit.cc]: @JasonGlazer Does RVariableTypes need to get allocated before this point? The test is failing on the CI build, and this is the only thing I can find that looks likely.

**Comment by JasonGlazer** [comment]
@jasondegraw Thanks for finding this issue, I wasn't seeing any problem when I was running it locally on my machine. I hope if fixes the CI builds.

**Comment by jasondegraw** [comment]
@JasonGlazer Edwin was the one who actually caught it, so I shouldn't take too much credit. The issue never came up on any of the platforms we tried manually, even Ubuntu - just on the CI.

I added a couple of deallocations of the globals in the test and merged in develop. Once the CI comes back I think we are good to go.

**Pull request merged**